File: Python3/ABC119/C.py
# URL: https://atcoder.jp/contests/abc119/tasks/abc119_c
# DFS

N, A, B, C = map(int,input().split())
L = [int(input()) for _ in range(N)]

def dfs(x, a, b, c):
  if(x == N):
    if(min(a, b, c) > 0):
      return abs(a-A)+abs(b-B)+abs(c-C)-30
    else:
      return 1e9+7
  ret0 = dfs(x+1, a, b, c)
  ret1 = dfs(x+1, a+L[x], b, c) + 10
  ret2 = dfs(x+1, a, b+L[x], c) + 10
  ret3 = dfs(x+1, a, b, c+L[x]) + 10
  return min(ret0, ret1, ret2, ret3)
print(dfs(0, 0, 0, 0))

File: C++/ABC119/C.cpp
#include <iostream>
#include <vector>

using namespace std;

#define rep(i,n) for(int i=0;i<n;++i)

int N, A, B, C;
int v[8];

int dfs(int x, int a, int b, int c){
  if (x == N){
    if (a == 0 || b == 0 || c == 0) return 1e9;
    return abs(a - A) + abs(b - B) + abs(c - C);
  }
  int ret = 1e9;
  ret = min(ret, dfs(x + 1, a, b, c));
  ret = min(ret, dfs(x + 1, a + v[x], b, c) + 10);
  ret = min(ret, dfs(x + 1, a, b + v[x], c) + 10);
  ret = min(ret, dfs(x + 1, a, b, c + v[x]) + 10);
  return ret;
}

int main(){
  cin >> N >> A >> B >> C;
  rep(i, 8) cin >> v[i];
  cout << dfs(0, 0, 0, 0) - 30 << endl;
}
